### PR TITLE
HAPI-397_skeleton

### PR DESCRIPTION
**Pull request by Sergei2404**

### CODE REVIEW

File: near/contract/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[package]
+name = "hapi-contract"
+authors = ["HAPI"]
+version = "0.1.0"
+edition = "2021"
+
+readme = "README.md"
+license = "GPL-3.0"
```
Comment (ars9) [inline near/contract/Cargo.toml:8]: Please use the same license as the root of the project: https://github.com/HAPIprotocol/hapi-core/blob/multichain/LICENSE

File: near/contract/build_docker.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+# Exit script as soon as a command fails.
+set -e
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+NAME="contract"
+
+if docker ps -a --format '{{.Names}}' | grep -Eq "^${NAME}\$"; then
```
Comment (ars9) [inline near/contract/build_docker.sh:10]: Please do not create a container like this. A proper docker build should be isolated. I.e. you need to create a Dockerfile that will set up the toolchain and targets, as well as fetch the needed cargo dependencies from the `Cargo.toml` and set up needed environment variables. 

The image would be built with a command like this:
```sh
docker build . -t $IMAGE_NAME
```

After such image is created, we can run it with a command similar to this:

```sh
docker run -it \
  -v $DIR:/builder
  $IMAGE_NAME
```

Please also check, whether we really need this line:
```
--cap-add=SYS_PTRACE --security-opt seccomp=unconfined
```

File: near/contract/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+[package]
+name = "hapi-contract"
```
Comment (ars9) [inline near/contract/Cargo.toml:2]: Please rename the package to "hapi-core-near".

**Review by ars9 — CHANGES_REQUESTED** [review]
Please add a `README.md` that would describe how to build and deploy this contract to a test environment.

File: near/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# Use NEAR maintained image as a base
+FROM nearprotocol/contract-builder:latest
+
+LABEL description="Container for build hapi-core-near contract"
```
Comment (ars9) [inline near/Dockerfile:4]: As we don't plan to distribute this image, we don't have to label it. And it's self-explanatory in the context.

File: near/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+# Define a volume for built artifacts and dependency cache
+VOLUME "/var/cache/cargo"
+
+# Define a volume for the output artifacts
+VOLUME "/output"
+
+# Build the contract when the container is started
+# Removing rlib first to reduce the size of wasm file
+CMD perl -i -pe 's/\["cdylib", "rlib"\]/\["cdylib"\]/' ./contract/Cargo.toml && \
```
Comment (ars9) [inline near/Dockerfile:31]: Do we need `rlib` in `Cargo.toml` at all?

File: near/contract/build_docker.sh (hunk 2 above)
Comment (ars9) [reply near/contract/build_docker.sh:10]: Why do we have "near/build_docker.sh" and "near/contract/build_docker.sh"?

**Comment by ars9** [comment]
There's still no information in `README.md` on how to build the contract (we have various scripts and cargo build - which ones to use in which cases?).

**Review by ars9 — CHANGES_REQUESTED** [review]

File: near/contract/build_docker.sh (hunk 2 above)
Comment (Sergei2404) [reply near/contract/build_docker.sh:10]: I will delete near/contract/build_docker.sh

File: near/Dockerfile (hunk 5 above)
Comment (Sergei2404) [reply near/Dockerfile:31]: Now no, but we will use it for tests

File: near/Dockerfile (hunk 4 above)
Comment (Sergei2404) [reply near/Dockerfile:4]: deleted

**Review by ars9 — APPROVED** [review]

**Pull request merged**